themes/bootstrap5/js/searchbox_controls.js

```diff
@@ -300,12 +300,13 @@ VuFind.register('searchbox_controls', function SearchboxControls() {
           },
           dataType: 'json',
           success: function autocompleteJSON(json) {
+            const regex = new RegExp(query.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&'), "ig");
```

Comment:
I think some comments might be helpful here:

```suggestion
            // Make the query regular-expression safe so we can do case-insensitive matching.
            // TODO: consider using RegExp.escape() here when it is more widely available.
            const regex = new RegExp(query.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&'), "ig");
```